src/main/java/org/spongepowered/common/world/schematic/SpongeSchematicBuilder.java

```diff
@@ -0,0 +1,158 @@
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.world.schematic;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.flowpowered.math.vector.Vector3i;
+import com.google.common.collect.Maps;
+import org.spongepowered.api.block.tileentity.TileEntityArchetype;
+import org.spongepowered.api.data.DataQuery;
+import org.spongepowered.api.data.DataView;
+import org.spongepowered.api.data.MemoryDataContainer;
+import org.spongepowered.api.world.extent.ArchetypeVolume;
+import org.spongepowered.api.world.extent.Extent;
+import org.spongepowered.api.world.extent.MutableBlockVolume;
+import org.spongepowered.api.world.schematic.BlockPalette;
+import org.spongepowered.api.world.schematic.BlockPaletteType;
+import org.spongepowered.api.world.schematic.BlockPaletteTypes;
+import org.spongepowered.api.world.schematic.Schematic;
+import org.spongepowered.api.world.schematic.Schematic.Builder;
+import org.spongepowered.common.SpongeImpl;
+import org.spongepowered.common.util.gen.ByteArrayMutableBlockBuffer;
+import org.spongepowered.common.util.gen.CharArrayMutableBlockBuffer;
+import org.spongepowered.common.util.gen.IntArrayMutableBlockBuffer;
+
+import java.util.Map;
+
+public class SpongeSchematicBuilder implements Schematic.Builder {
+
+    private ArchetypeVolume volume;
+    private Extent view;
+    private BlockPalette palette;
+    private BlockPaletteType type = BlockPaletteTypes.LOCAL;
+    private DataView metadata;
+    private Map<String, Object> metaValues = Maps.newHashMap();
+
+    @Override
+    public Builder volume(ArchetypeVolume volume) {
+        this.volume = volume;
+        return this;
+    }
+
+    @Override
+    public Builder volume(Extent volume) {
+        this.view = volume;
+        return null;
```

Comment:
Why does this method return `null` (ln. 68) ? I think this is an error as the javadoc says :

> Returns This builder, for chaining

Fortunately it's not too hard to find a work around : 
```
Builder b = Schematic.builder();
b.volume(extent);
schematic = b.build();
```
:wink: